diff --git a/docker/local.dockerfile b/docker/local.dockerfile
index 8850acb..2602841 100644
--- a/docker/local.dockerfile
+++ b/docker/local.dockerfile
@@ -37,6 +37,15 @@ set -ex
 set -o pipefail
 trap 'echo "Exit status $? at line $LINENO from: $BASH_COMMAND"' ERR
 
+echo "deb http://snapshot.debian.org/archive/debian-ports/20250801T194242Z/ sid main" > /etc/apt/sources.list
+( echo 'quiet "true";'; \
+  echo 'APT::Get::Assume-Yes "true";'; \
+  echo 'APT::Install-Recommends "false";'; \
+  echo 'Acquire::Check-Valid-Until "false";'; \
+  echo 'Acquire::Retries "5";'; \
+) > /etc/apt/apt.conf.d/99cpython-portable
+rm -f /etc/apt/sources.list.d/*
+
 apt-get update
 
 DEBIAN_FRONTEND=noninteractive apt-get install -y curl python3.12 \
@@ -53,6 +62,8 @@ EOF
 # preload tiktoken
 ENV TIKTOKEN_CACHE_DIR=/app/.tiktoken
 
+ARG PIP_EXTRA_INDEX_URL
+ARG UV_EXTRA_INDEX_URL
 # Install dify_plugin to speedup the environment setup, test uv and preload tiktoken
 RUN <<EOF bash
 
diff --git a/pkg/entities/constants/arch.go b/pkg/entities/constants/arch.go
index 3e5ab55..b05358d 100644
--- a/pkg/entities/constants/arch.go
+++ b/pkg/entities/constants/arch.go
@@ -10,11 +10,12 @@ type Arch string
 const (
 	AMD64 Arch = "amd64"
 	ARM64 Arch = "arm64"
+	LOONG64 Arch = "loong64"
 )
 
 func isAvailableArch(fl validator.FieldLevel) bool {
 	value := fl.Field().String()
-	return value == string(AMD64) || value == string(ARM64)
+	return value == string(AMD64) || value == string(ARM64) || value == string(LOONG64)
 }
 
 func init() {
